TACC/Core-CMS
# GH-75: Hotfix: Add Missing DB Migration (Small)

# Overview

Add missing migration for change of description of field for label.

# Changes

- Change description to not mention how to make a link (because a bigger description is in the "Link" section.

# Issue

- GH-75

# Screenshots

![GH-75 Hotfix](https://user-images.githubusercontent.com/62723358/130626564-ff5d8049-d555-4d40-881e-8a995d167ffc.png)

# Testing

1. In CMS container, run `python manage.py makemigrations`.

    <details>
    <summary>Output</summary>

    ```
    root@wes_core_cms:/code# python manage.py makemigrations
    Checking for secret production values
    Production secrets found, using values
    Migrations for 'taccsite_data_list':
    taccsite_cms/contrib/taccsite_data_list/migrations/0002_auto_20210824_0803.py
        - Alter field key on taccsitedatalistitem
    ```

    </details>

2. Check that a migration is made i.e. output is as described above.
3. In CMS container, run `python manage.py migrate`.

    <details>
    <summary>Output</summary>

    ```
    root@wes_core_cms:/code# python manage.py migrate
    Checking for secret production values
    Production secrets found, using values
    Operations to perform:
    Apply all migrations: admin, auth, bootstrap4_alerts, bootstrap4_badge, bootstrap4_card, bootstrap4_carousel, bootstrap4_collapse, bootstrap4_content, bootstrap4_grid, bootstrap4_jumbotron, bootstrap4_link, bootstrap4_listgroup, bootstrap4_media, bootstrap4_picture, bootstrap4_tabs, bootstrap4_utilities, cms, contenttypes, djangocms_column, djangocms_file, djangocms_googlemap, djangocms_icon, djangocms_link, djangocms_picture, djangocms_snippet, djangocms_style, djangocms_text_ckeditor, djangocms_video, easy_thumbnails, filer, menus, sessions, sites, taccsite_data_list, taccsite_sample, taccsite_system_monitor
    Running migrations:
    Applying taccsite_data_list.0002_auto_20210824_0803... OK
    root@wes_core_cms:/code# 
    ```

    </details>

4. Check that Data List Item plugin's label field description is shorter.

# Notes

The link plugin was not extended and its fields borrowed to be able to create a link directly within a Data List Item plugin (as was done for other plugins that link), because:
- the Data List Item __value__ field may support a link
    - _So would embedded Link field belong to label or value?_\*
- the Data List Item __value__ field may support a link _within text_
    - _Value would need child plugin (["Link in Text"](https://github.com/TACC/Core-CMS/issues/306)), so it would be consistent to support Link for label also via child plugin._\*
- the Data List Item __label__ field may support a link _within text_
    - _Label would need child plugin (["Link in Text"](https://github.com/TACC/Core-CMS/issues/306)), so it would be consistent to support Link for label also via child plugin._\*

> \* As with all child plugin solutions in any of our plugins, if we learn how to provide plugin fields such that child plugins are not necessary, then supporting child plugins can be a legacy feature when a new way is added.

<details>
<summary>From Where Do These Possibilities Come?</summary>

These possibilities were imagined by Dev, and were confirmed or noted as possible by Design.

> For [Frontera System page] use of a data list, and beyond
> 1. Is __value__ allowed to be a link?
> 2. Is __value__ allowed to have a link within its text?
> 3. Is __value__ allowed to be an image?
> 4. Is __value__ allowed to have an image within its text?
> 5. Is __label__ allowed to have a link within its text?
> 6. Is __label__ as link allowed to be bold?
>
> — @tacc-wbomar

> 1, no, 2. I don't think so, 3. maybe, but doubt it, 4. no. 5. yes. 6, maybe
>
> — M.S. (designer)

</details>
